src/Core/KeyManagement/Queries/CanUseKeyConnectorQuery.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Bit.Core.Context;
+using Bit.Core.Entities;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
+using Bit.Core.KeyManagement.Queries.Interfaces;
+
+namespace Bit.Core.KeyManagement.Queries;
+
+public class CanUseKeyConnectorQuery : ICanUseKeyConnectorQuery
+{
+    private readonly ICurrentContext _currentContext;
+
+    public CanUseKeyConnectorQuery(ICurrentContext currentContext)
+    {
+        _currentContext = currentContext;
+    }
+
+    public void VerifyCanUseKeyConnector(User user)
```

Comment:
💭  Perhaps this would be better as an AuthorizationHandler on the controller. It is odd for a query not to return any data. Happy to help if you would like to see what that would look like.